src/transport/net.cc

```diff
@@ -1489,13 +1589,20 @@ static ncclResult_t recvProxyProgress(struct ncclProxyState* proxyState, struct
         if (done) {
           for (int i=0; i<subGroup->groupSize; i++) {
             struct ncclProxySubArgs* sub = subGroup + i;
+
             sub->transmitted += args->sliceSteps;
-            for (uint64_t step=sub->transmitted-args->sliceSteps; step<sub->transmitted; step++) ncclProfilingRecord(args, s+i, step, ncclProxyProfileRecvGPUWait);
+            ncclProfilerRecordProxyOpEventState(s+i, args, sub->transmitted, sub->transSize, ncclProfilerProxyOpRecvTransmitted);
+            ncclProfilerRecordProxyStepEventStates(s+i, args, sub->transmitted-args->sliceSteps, sub->transmitted, ncclProfilerProxyStepRecvGPUWait);
             if (step < sub->nsteps) {
               __sync_synchronize();
               struct recvNetResources* resources = (struct recvNetResources*) (sub->connection->transportResources);
               volatile uint64_t* recvTail = resources->gdcSync ? resources->gdcSync : &resources->recvMem->tail;
-              *recvTail = sub->base + sub->transmitted;
+              if (sub->reg) {
+                // We may have added more net steps, but reg operations only have a single step w.r.t. the GPU.
+                if (sub->transmitted == sub->nsteps) *recvTail = sub->base + args->sliceSteps;
+              } else {
```

Comment:
In NCCL 2.23 it's
```
else
    *recvTail = sub->base + sub->transmitted;
````
instead of 
```
else {
    *recvTail = sub->base + sub->transmitted;
}
````
need recover it to avoid non-MSCCL changes included.

Reply:
Resolved in latest commit.